### PR TITLE
Support filtering to the TargetGroup property value

### PR DESCRIPTION
**Pull request by tarekgh**

### CODE REVIEW

**Comment by tarekgh** [comment]
@weshaggard please have a look. 
CC @MattGal

File: dir.traversal.targets [hunk 1]
```diff
@@ -47,6 +53,13 @@
       <Project Remove="@(Project)" />
       <Project Include="@(ProjectsToBuild)" />
     </ItemGroup>
+
+    <ItemGroup Condition="'$(FilterToTargetGroup)'!=''">
+      <FilteredProjectsToBuild Include="@(Project)" Condition="'%(Project.TargetGroup)'=='$(FilterToTargetGroup)' Or '%(Project.Extension)' == '.proj' Or '%(Project.Extension)' == '.builds'" />
```
Comment (weshaggard) [inline dir.traversal.targets]: Can you change this to something similar to what we are doing with OSGroup in PR https://github.com/dotnet/corefx/pull/7535/files#diff-cd0fc5e0bad8102e1a45aa7575bdd102R53.

**Comment by weshaggard** [comment]
Looks good beyond the one comment. I think switching how we do that condition will help us with allowing the filtering for multiple version of a given TFM.

**Comment by tarekgh** [comment]
@weshaggard I have fixed the issue to support a list of TargetGroup

File: src/System.Globalization.Calendars/tests/project.json [hunk 2]
```diff
@@ -14,6 +14,11 @@
   "frameworks": {
     "dnxcore50": {
       "imports": "portable-net45+win8"
-    }
+    },
+    "net46": {
```
Comment (weshaggard) [inline src/System.Globalization.Calendars/tests/project.json]: cc @chcosta here is an example of a test project.json file that has multiple dependency sections.

**Pull request merged**